From 5e0e83e5d77f145c945f318ccd2152e0cc64c0d2 Mon Sep 17 00:00:00 2001
From: Anson Huang <Anson.Huang@nxp.com>
Date: Fri, 20 Oct 2017 23:42:24 +0800
Subject: [PATCH 2669/5242] MLK-16676-1 soc: imx: pm-domains: support multiple
 early_power_on resource

commit  8b00fa404a166185c0a075e82896b2140c165c29 from
https://source.codeaurora.org/external/imx/linux-imx.git

On i.MX8QM/i.MX8QXP, there could be multiple resources
need to be powered on earlier after resume, current variable
of index could be reset for different power domain nodes and
cause resource id overwrite issue, fix the array index type
to support multiple early power on case.

Signed-off-by: Anson Huang <Anson.Huang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/soc/imx/pm-domains.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/soc/imx/pm-domains.c b/drivers/soc/imx/pm-domains.c
index a99883a..a3bd4f2 100644
--- a/drivers/soc/imx/pm-domains.c
+++ b/drivers/soc/imx/pm-domains.c
@@ -234,7 +234,7 @@ static int __init imx8_add_pm_domains(struct device_node *parent,
 					struct generic_pm_domain *genpd_parent)
 {
 	struct device_node *np;
-	int index = 0;
+	static int index;
 
 	for_each_child_of_node(parent, np) {
 		struct imx8_pm_domain *imx8_pd;
-- 
1.7.9.5

